src/Search/CitizenProjectSearchResultsProvider.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace AppBundle\Search;
+
+use AppBundle\CitizenProject\CitizenProjectManager;
+use AppBundle\Repository\CitizenProjectRepository;
+
+class CitizenProjectSearchResultsProvider implements SearchResultsProviderInterface
+{
+    private $citizenProjectRepository;
+    private $citizenProjectManager;
+
+    public function __construct(CitizenProjectRepository $citizenProjectRepository, CitizenProjectManager $citizenProjectManager)
+    {
+        $this->citizenProjectRepository = $citizenProjectRepository;
+        $this->citizenProjectManager = $citizenProjectManager;
+    }
+
+    public function find(SearchParametersFilter $search): array
+    {
+        $result = $this->citizenProjectRepository->searchAll($search);
+
+        foreach ($result as $citizenProject) {
+            $citizenProject->setCreator($this->citizenProjectManager->getCitizenProjectCreator($citizenProject));
```

Comment:
This is fu.... weird. I know you've already asked before, but why is this needed in the first place. We should maybe revise this model.

Reply:
yes... I spent 1 or 2 hours looking for a decent solution but I was not able to find one reasonable..

Reply:
Damned...

Reply:
Is there any way to change the default query performed by the EntityManager ?

Reply:
Because I find something cool by mapping the creator field into Doctrine but it requires to use a specific DQL each time we need to retrieve the entity.

Reply:
Can this DQL injection be abstracted in some base repository class or a trait?

Reply:
For sure it could be